### PR TITLE
Adding a warning whenever non-scalable discovert_parts_snd is triggered

### PR DESCRIPTION
**Pull request by amartinhuertas**
See issue https://github.com/fverdugo/PartitionedArrays.jl/issues/41

### CODE REVIEW

File: src/Interfaces.jl [hunk 1]
```diff
@@ -495,8 +495,26 @@ function discover_parts_snd(
   parts_snd
 end
 
+function _warn_message_on_main_task_discover_parts_snd(data::AbstractPData)
+   map_parts(get_part_ids(data)) do part
```
Comment (fverdugo) [inline src/Interfaces.jl:499]: Use function `map_main` if you only want to map the task on the first task

File: src/SequentialBackend.jl [hunk 2]
```diff
@@ -198,3 +198,7 @@ function async_exchange!(
   end
   t_out
 end
+
+# Do not generate warning if SequentialBackend
+function _warn_message_on_main_task_discover_parts_snd(data::SequentialData)
```
Comment (fverdugo) [inline src/SequentialBackend.jl:203]: I think we also want the warning in the sequential back-end. We want to discover this warning as soon as possible. Better when we still are in sequantial mode

**Review by fverdugo — CHANGES_REQUESTED** [review]

File: src/Interfaces.jl (hunk 1 above)
Comment (amartinhuertas) [reply src/Interfaces.jl]: Ok. Good to know such function exists. Did not know. Solved now.

File: src/SequentialBackend.jl (hunk 2 above)
Comment (amartinhuertas) [reply src/SequentialBackend.jl]: Ok. Agreed. Makes sense. Done.

**Review by fverdugo — APPROVED** [review]

**Pull request merged**